### PR TITLE
Add custom fields tagging to the product

### LINKED ISSUE #175: Some of the attributes are not tagged in the sku custom fields

### PR DESCRIPTION
**Pull request by chenliangde**
Closes #175

### CODE REVIEW

File: etc/config.xml [hunk 1]
```diff
@@ -9,6 +9,7 @@
             </images>
             <flags>
                 <variation_tagging>1</variation_tagging>
+                <use_custom_fields>0</use_custom_fields>
```
Comment (mridang) [inline etc/config.xml]: Enable this by default. I think that would be a better approach.

File: Model/Product/Sku/Builder.php [hunk 2]
```diff
@@ -153,4 +159,70 @@ public function buildImageUrl(Product $product, Store $store)
...
     }
+
+    /**
+     * Tag the custom attributes
+     *
+     * @param Product $product
+     * @param NostoSku $nostoSku
+     * @param Store $store
+     */
+    private function loadCustomFieldsFromConfigurableAttributes(Product $product, NostoSku $nostoSku, Store $store)
```
Comment (hannunosto) [inline Model/Product/Sku/Builder.php]: How about putting this into trait? It's almost (only the NostoSku / NostoProduct differs) identical with this https://github.com/Nosto/nosto-magento2/pull/201/files#diff-c7bb4cf85d1abbf4cf4b2d1cec60f45aR406

File: etc/adminhtml/system.xml [hunk 3]
```diff
@@ -101,6 +101,13 @@
                     </comment>
                     <source_model>Magento\Config\Model\Config\Source\Yesno</source_model>
                 </field>
+                <field id="use_custom_fields" translate="label comment" type="select" sortOrder="10"
+                       showInDefault="1" showInWebsite="1" showInStore="1">
+                    <label>Enable custom fields</label>
+                    <comment>Send attributes from product attribute set to nosto as custom fields
```
Comment (hannunosto) [inline etc/adminhtml/system.xml]: nosto -> Nosto

File: Model/Product/Builder.php [hunk 4]
```diff
@@ -52,11 +55,14 @@
 use Nosto\Tagging\Model\Product\Sku\Collection as NostoSkuCollection;
 use Nosto\Tagging\Model\Product\Url\Builder as NostoUrlBuilder;
 use Nosto\Tagging\Model\Product\Tags\LowStock as LowStockHelper;
+use Nosto\Object\Product\Product as NostoProduct;
 use Nosto\Types\Product\ProductInterface;
 use Nosto\Tagging\Logger\Logger as NostoLogger;
 
 class Builder
 {
+    const ATTRIBUTE_VALUE_ANY = '--ANY--';
```
Comment (hannunosto) [inline Model/Product/Builder.php]: Is this used anywhere?

**Pull request merged**